### PR TITLE
fix: remove automated contribution message

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

Unfortunately, the automated contribution message does not work properly. To evaluate this correctly, we need to first filter out all merge commits. To do this, we need to check the length of the `parents` property. I did not find a way to do this with jinja2 filters. Which would allow us to do something like `rejectattr`.

In the code snippet removed in this PR, I attempted to create a property to filter the merge commits on. This also doesn't work because the underlying object exposed by mergify is not a dictionary but a class and thus does not have the `update` function.

Mergify is shipping an additional property next month: https://github.com/Mergifyio/mergify/discussions/4636#discussioncomment-6295474. Thus, I am replacing this automated mechanism with a manual section that we can add to the PR until mergify ships the additional property that we can utilize to implement this in a clean way.

Related: #4130.
Related: #4104.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by thomaseizinger** [comment]
cc @mxinden

File: .github/mergify.yml [hunk 1]
```diff
@@ -8,19 +8,8 @@ defaults:
...
-          {%- else -%}
-            {%- set _ = commit.update({'merge': false}) -%}
-          {%- endif -%}
-        {%- endfor -%}
-        {%- for commit in (commits | rejectattr("merge") | unique(False, 'email_author')) | rejectattr("author", "==", author) -%}
-        Co-authored-by: {{ commit.author }} <{{ commit.email_author }}>
-        {% endfor %}
-        {# GitHub requires that the `Co-authored-by` lines are AT THE VERY END of a commit, hence nothing must come after this. #}
+
+        {{ body | get_section("## Attributions", "") }}
```
Comment (mxinden) [inline .github/mergify.yml:12]: Want to make this part of our pull request template?

https://github.com/libp2p/rust-libp2p/blob/master/.github/pull_request_template.md

**Review by mxinden — COMMENTED** [review]
Thanks for looking so deep into this!

File: .github/mergify.yml (hunk 1 above)
Comment (thomaseizinger) [reply .github/mergify.yml:12]: It is only temporary so I am not too fussed.

**Pull request merged**